packages/components/src/unit-control/index.tsx

```diff
@@ -200,7 +250,7 @@ function UnforwardedUnitControl(
 		<Root className="components-unit-control-wrapper" style={ style }>
 			<ValueInput
 				aria-label={ label }
-				type={ isPressEnterToChange ? 'text' : 'number' }
+				type="text"
```

Comment:
From the [initial conversation](https://github.com/WordPress/gutenberg/pull/39303#discussion_r824039292) around this change, I initially though that we were considering to change the `type` to always be `number` in order to enjoy native browser validation.

This PR is taking the opposite approach (always set the `type` to `text`) — I'm just wondering if this change is going to impact the way users interact with the component (since the native browser validation is never going to be enforced).

If we decide to go down this route, I agree with @stokesman and @mirka that we'd really need to be careful with how the new UX works, and add as many unit tests as possible.

Reply:
> If there were a strong reason to keep the `type=number` the paste handling could be simplified to replace the whole input value anytime a unit is included in the paste. It might even be more convenient that way but you'd have to discover that behavior and it's unconventional.

Would it be hard to experiment with always having the input being `type="number"` ? Would that also mean that we'd need to remove the `isPressEnterToChange` functionality?